src/libmongoc/tests/test-mongoc-structured-log.c

```diff
@@ -714,4 +821,11 @@ test_structured_log_install (TestSuite *suite)
    TestSuite_Add (suite, "/structured_log/duration", test_structured_log_duration);
    TestSuite_Add (suite, "/structured_log/level_names", test_structured_log_level_names);
    TestSuite_Add (suite, "/structured_log/component_names", test_structured_log_component_names);
+   TestSuite_Add (suite, "/structured_log/max_document_length", test_structured_log_max_document_length);
```

Comment:
```suggestion
   TestSuite_AddFull (suite,
                      "/structured_log/max_document_length",
                      test_structured_log_max_document_length,
                      NULL,
                      NULL,
                      check_test_structured_log_env_defaults);
```

Register other structured logging tests which may be impacted by env vars at runtime with the same skip condition used for the `/structured_log/env_defaults` test below.

Reply:
It should be fine as it, I moved the default test for max_document_length into test_structured_log_env_defaults(). test_structured_log_max_document_length() is just the remaining tests which don't depend on env settings.

Reply:
Given the env var is read on every construction of the `mongoc_structured_log_opts_new` object, all tests which construct the object may be affected, e.g.:
```
$ MONGODB_LOG_MAX_DOCUMENT_LENGTH=1 ./test-libmongoc -d -l "/structured_log/json"
...
Begin /structured_log/json, seed 1057888386
FAIL

Assert Failure:
  "{ "message" : "Log entry with deferred BSON-to-JSON", "kJSON" : "{...", "kNull" : null }"
  !=
  "{ "message" : "Log entry with deferred BSON-to-JSON", "kJSON" : "{ \"k\" : \"v\" }", "kNull" : null }"
 src/libmongoc/tests/test-mongoc-structured-log.c:54  structured_log_func()
```
```
$ MONGODB_LOG_MAX_DOCUMENT_LENGTH=1 ./test-libmongoc -d -l "/structured_log/command"
...
Begin /structured_log/command, seed 897720391
FAIL

Assert Failure:
  "{ "message" : "Log entry with command and reply fields", "commandName" : "Not a command", "databaseName" : "Some database", "commandName" : "Not a command", "operationId" : 1311768468877410321, "command" : "{...", "reply" : "{...", "reply" : "{...", "reply" : "{}", "failure" : { "r" : "s", "code" : 1 }, "failure" : { "r" : "s", "code" : 1 }, "failure" : { "code" : 1 }, "failure" : { "code" : 123, "domain" : 456, "message" : "oh no" } }"
  !=
  "{ "message" : "Log entry with command and reply fields", "commandName" : "Not a command", "databaseName" : "Some database", "commandName" : "Not a command", "operationId" : 1311768468877410321, "command" : "{ \"c\" : \"d\", \"first_payload\" : [ { \"i\" : 0, \"x\" : 0 }, { \"i\" : 0, \"x\" : 1 }, { \"i\" : 0, \"x\" : 2 }, { \"i\" : 0, \"x\" : 3 }, { \"i\" : 0, \"x\" : 4 } ], \"second_payload\" : [ { \"i\" : 1, \"x\" : 0 }, { \"i\" : 1, \"x\" : 1 }, { \"i\" : 1, \"x\" : 2 }, { \"i\" : 1, \"x\" : 3 }, { \"i\" : 1, \"x\" : 4 } ] }", "reply" : "{ \"r\" : \"s\", \"code\" : 1 }", "reply" : "{ \"r\" : \"s\", \"code\" : 1 }", "reply" : "{}", "failure" : { "r" : "s", "code" : 1 }, "failure" : { "r" : "s", "code" : 1 }, "failure" : { "code" : 1 }, "failure" : { "code" : 123, "domain" : 456, "message" : "oh no" } }"
 src/libmongoc/tests/test-mongoc-structured-log.c:54  structured_log_func()
```
```
$ MONGODB_LOG_ALL=off ./test-libmongoc -d -l "/structured_log/opts"
...
Begin /structured_log/opts, seed 3267837868
FAIL

Assert Failure: 4 == 0
src/libmongoc/tests/test-mongoc-structured-log.c:72  test_structured_log_opts()
```

Reply:
It's not appropriate to skip these tests. These tests should be modified not to depend on the default.

In general, I'm happy to guard against environment settings that conflict with the actual test being performed, but it's not reasonable to exhaustively list environment settings that would cause the test to fail.